test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -126,6 +130,36 @@ public void testStreamContents(InputStreamProvider inputStreamProvider,
         checkTransferredContents(inputStreamProvider, outputStreamProvider, createRandomBytes(4096, 0), 0, 4096);
     }
 
+    /*
+     * Special test for file-to-file transfer of more than two GB.
+     * This test covers multiple iterations of FileChannel.transerTo(FileChannel),
+     * which ChannelInputStream.transferTo() only applies in this particular case,
+     * and cannot get tested using a single byte[] due to size limitation of arrays.
+     */
+    @Test
+    public void testMoreThanTwoGB() throws IOException {
+        // preparing two temporary files which will be compared at the end of the test
+        Path sourceFile = Files.createTempFile(null, null);
+        Path targetFile = Files.createTempFile(null, null);
+
+        // writing 3 GB of random bytes into source file
+        for (int i = 0; i < NUM_WRITES; i++)
+            Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
+
+        // performing actual transfer, effectively by multiple invocations of Filechannel.transferTo(FileChannel)
+        long count;
+        try (InputStream inputStream = Channels.newInputStream(FileChannel.open(sourceFile));
+        OutputStream outputStream = Channels.newOutputStream(FileChannel.open(targetFile, StandardOpenOption.WRITE))) {
```

Comment:
I think that this needs to be indented.

Reply:
Thanks, done in https://github.com/openjdk/jdk/pull/6093/commits/435690c6539e74b08162b63d0b896545c5ee3de2.